library/src/main/java/com/novoda/downloadmanager/DownloadBatch.java

```diff
@@ -86,31 +91,34 @@ void download() {
         deleteBatchIfNeeded(downloadBatchStatus, downloadsBatchPersistence, callback);
         notifyCallback(callback, downloadBatchStatus);
         callbackThrottle.stopUpdates();
-        Log.v("end download " + rawBatchId);
+        Log.v("end sync download " + rawBatchId);
     }
 
     private static boolean shouldAbortStartingBatch(ConnectionChecker connectionChecker,
                                                     DownloadBatchStatusCallback callback,
                                                     InternalDownloadBatchStatus downloadBatchStatus,
                                                     DownloadsBatchPersistence downloadsBatchPersistence) {
-        DownloadBatchStatus.Status status = downloadBatchStatus.status();
-
-        if (status == DELETED) {
+        // WARNING: do not extract downloadBatchStatus.status() as a local variable, this will
```

Comment:
Happens in a `thread` worth highlighting with an annotation? I'm not a fan of comments 😄